tensorflow/python/data/util/sparse_test.py

```diff
@@ -18,58 +18,405 @@
 from __future__ import division
 from __future__ import print_function
 
+import functools
+
+from absl.testing import parameterized
+
+from tensorflow.python.data.kernel_tests import test_base
 from tensorflow.python.data.util import nest
 from tensorflow.python.data.util import sparse
+from tensorflow.python.framework import combinations
 from tensorflow.python.framework import constant_op
 from tensorflow.python.framework import dtypes
 from tensorflow.python.framework import ops
 from tensorflow.python.framework import sparse_tensor
 from tensorflow.python.framework import tensor_shape
-from tensorflow.python.framework import test_util
 from tensorflow.python.platform import test
 
 
-class SparseTest(test.TestCase):
-
-  def testAnySparse(self):
-    test_cases = (
-        {
-            "classes": (),
-            "expected": False
-        },
-        {
-            "classes": (ops.Tensor),
-            "expected": False
-        },
-        {
-            "classes": (((ops.Tensor))),
-            "expected": False
-        },
-        {
-            "classes": (ops.Tensor, ops.Tensor),
-            "expected": False
-        },
-        {
-            "classes": (ops.Tensor, sparse_tensor.SparseTensor),
-            "expected": True
-        },
-        {
-            "classes": (sparse_tensor.SparseTensor, sparse_tensor.SparseTensor),
-            "expected":
-                True
-        },
-        {
-            "classes": (sparse_tensor.SparseTensor, ops.Tensor),
-            "expected": True
-        },
-        {
-            "classes": (((sparse_tensor.SparseTensor))),
-            "expected": True
-        },
+# NOTE(vikoth18): Arguments of parameterized tests are lifted into lambdas to make
+# sure they are not executed before the (eager- or graph-mode) test environment
+# has been set up.
+#
+
+def _test_any_sparse_combinations():
+
+  cases = [
+      ("TestCase_0", lambda: (), False),
+      ("TestCase_1", lambda: (ops.Tensor), False),
+      ("TestCase_2", lambda: (((ops.Tensor))), False),
+      ("TestCase_3", lambda: (ops.Tensor, ops.Tensor), False),
+      ("TestCase_4", lambda: (ops.Tensor, sparse_tensor.SparseTensor), True),
+      ("TestCase_5", lambda: (sparse_tensor.SparseTensor, sparse_tensor.SparseTensor), True),
+      ("TestCase_6", lambda: (((sparse_tensor.SparseTensor))), True)
+  ]
+  def reduce_fn(x, y):
+    name, classes_fn, expected = y
+    return x + combinations.combine(
+        classes_fn=combinations.NamedObject(
+            "classes_fn.{}".format(name), classes_fn
+        ),
+        expected=combinations.NamedObject(
```

Comment:
I don't think `expected` needs to be wrapped in a NamedObject here (what benefit does it provide?). Not making it a `NamedObject` avoids the need for `_obj` later.

Reply:
addressed in the commit.